src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -10666,9 +10666,7 @@ void __stdcall ProfilerUnmanagedToManagedTransitionMD(MethodDesc *pMD,
 // These do a lot of work for us, setting up Frames, gathering arg info and resolving generics.
   //*******************************************************************************************
 
-HCIMPL2_RAW(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecificHandle)
-GCX_COOP_THREAD_EXISTS(GET_THREAD());
-HCIMPL_PROLOG(ProfileEnter)
+HCIMPL2(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecificHandle)
```

Comment:
Should we fix `ProfileLeave` to look the same?

Reply:
Is it possible that the ProfileLeave would get called without the ProfileEnter? It seemed to me that the ProfileEnter would setup the thread and ProfileLeave would then always have it set up already.

Reply:
Right.

I meant that ProfileLeave looks like this currenly:

```
HCIMPL2_RAW(EXTERN_C void, ProfileLeave, UINT_PTR clientData, void * platformSpecificHandle)
GCX_COOP();
HCIMPL_PROLOG(ProfileLeave)
```

Should it be changed to just this:

`HCIMPL2(EXTERN_C void, ProfileLeave, UINT_PTR clientData, void * platformSpecificHandle)`

Reply:
Ah, so we would move the GCX_COOP inside of the body, makes sense.

Reply:
Ah, I guess the GCX_COOP is not needed as the ProfileLeave is always called in coop mode?